# Can Reading Ones Genome Provide Information On Who They Are: An Investigation of Eye Color Using GWAS and Mathematical Modeling. 
##### Mauro Chavez
##### 5/7/2018

## Abstract 

This week involved the analysis of GWAS data to identify significant genetic markers in determining an individual’s eye color. Additionally, mathematical modeling of an individual’s alleles at specific loci was completed to provide probabilities that a person had a given eye color. Identified SNPs and provided SNPs were studied in reference genomes, in the end highlighting SLC24A4 and OCA2 genes as potentially playing a major role in determining eye color. This type of analysis is important because when applied to health, one can begin to generate information on susceptibility to disease based on specific genetic markers. 

## Introduction

Lab this week came in two parts with two different goals. The first goal was to work through a GWAS pipeline on DNA samples. Using SNP profiles for samples in the data set, the hope was to filter out the significant variants that potentially play a role in determining an individuals eye color. The second goal involved using a smaller data set of 6 SNPs to build a mathematical model that could be used to predict an individuals eye color based off their alleles at 6 specific locations, the significance of which was determined by the IrisPlex paper [9]. The overarching question being, with genomic data on an individual’s SNPs, can the color of their eyes be determined by looking at the right places?

Answering this question is important because once one knows how to use GWAS as a tool, applying it to different contexts can provide extremely useful information on traits involved with disease and human health. Mccarthy Et al explain in their paper, “The first wave of large-scale, high-density genome-wide association (GWA) studies has improved our understanding of the genetic basis of many complex traits. For several diseases, including type 1 (Refs 2,3) and type 2 diabetes, inflammatory bowel disease, prostate cancer and breast cancer, there has been rapid expansion in the numbers of loci implicated in predisposition” [5]. While this comments on how the analysis of GWA studies can provide critical information that allows us to make smarter choices in life based on what diseases we might be at risk for, the paper itself discusses the areas in which GWAS needs further development. While GWAS requires more discussion on topics like selecting cases, selecting controls, designing SNP array chips, deciding which SNPs to survey and which to assume by correlation, the skills learned in this lab will only become more valuable as better data is provided by researchers. 

Answering the questions mentioned above was split into two parts. The first involved finding the SNPs that may be associated with eye color. First, or data was turned into two sets, an unprocessed set and one that underwent Principal Component Analysis. These two data sets were surveyed to determine which one provided better information on significant variants. From there, variants were clumped together to identify the important genetic signals associated with eye color. The second part involved mathematical modeling and statistical analysis to predict eye color based on allele presence. In all stages of analysis, reference genomes were consulted to provide context on where SNPs were located and what functional role they might play. 

## Methods 

### Datasets

This lab included two data sets, one for the GWAS portion and one for use in the eye color prediction model. For the GWAS completed in this lab, DNA samples were gathered and eye color recorded for 261 subjects from different ethnic backgrounds. In this group of 261 people, 104 had brown eyes giving a brown eye frequency of 0.398 and 157 had blue eyes, giving a blue eye frequency of 0.602. Samples were exposed to an SNP genotype array. Resulting data had duplicated samples and problematic markers removed for cleaning. From each sample's DNA, data was collected on 911774 variants or SNPs. The samples used in the color prediction model were collected independently from the samples used in the GWAS portion. This smaller set contained data from 206 subjects with 107 coming from Southern Europe (Labeled TSI), and 99 coming from Northern Europe (labeled CEU) in .vcf file format. bcftools was used to process the VCF file into a more usable format [2]. The file was transposed making each sample its own column so that an “ID” field could be appended as the first column. Genotype data for each SNP was organized to be separated by tabs, and `|` characters were removed. After processing, the file was transposed again to make samples appear as rows. Alleles for each individual were collected for 6 SNPs (rs12913832, rs1800407, rs12896399, rs16891982, rs1393350, and rs12203592). 

### GWAS
Plink version v1.90b5.4 was used with `--pca 10` flag to complete Principal Component Analysis (PCA), identifying the most significant features that best explain the variation present in the data [1]. The results of PCA by Plink were used as a set of covariates for GWAS. Python was used to generate a scatter plot of PC1 vs PC2, the first two identified principal components. Code for plot generation is located in scripts folder and called makeScatPlot.py. The GWAS itself was completed using plink with `--logistic` to signify use of logistic regression and `--allow-no-sex` flags on both the covariate and pre-PCA data sets. ` --covar` flag was included for GWAS on covariate data set. Running GWAS on the principal components controlled for population structure by removing SNPs more likely associated with differences in ethnic backgrounds rather than eye color. The results of logistic regression on principal components provided better data as confirmed by QQ and Manhattan plot generation, and were used for clumping. gwas_plotter.py in scripts folder was used for generation of these plots. In clumping, Plink was used to merge causal variants with the nearby variants they were correlated with, producing independent signals.  Plink was run with index variant p-value threshold set to 0.0001, clumped variant p-value threshold set to 0.01, r^2 threshold set to 0.5 and clump kb radius set to 250. This means that locations within 250 kb from an index variant, with r^2 greater than 0.5 and association p-value less than 0.01 to that index variant are included in that variant’s clump. Index variants were picked by identifying variants with p-value less than or equal to 0.0001. Clumps with p<5*10^-8 were reported as meeting genome-wide significance. 

### Eye color prediction
Eye color prediction on the second data set was completed using a Google spreadsheet. Each sample was loaded in as a row with ID, population label, and genotype for each SNP listed in separate columns. The following 6 columns counted the minor alleles the individual carried, and the next 4 completed calculations. The first calculation completed the sum of the product of beta1’s and associated minor allele count, the second being the same except using beta2’s. The third and fourth computed e raised to the power of alpha1+first calculation column and e raised to alpha2+second calculation column. These pieces allowed for the first 3 columns of each row to combine these calculations in order to generate probability blue/other/brown eye color, with the 4th reporting the color with highest probability. To interpret the function of these 6 variants, UCSC Genome Browser was used to view human reference genome hg19 with ENCODE histone modification, Integrated Regulation from ENCODE, and Expression QTLs in 44 tissues from GTEx tracks shown [3]. Findings from SNP interpretation verified by checking snpedia.com database [4]

## Results

### GWAS
After initial Principal Component Analysis, principal component 1 and principal component 2 were graphed in a scatter plot using python as shown in Fig 1. The generation of three distinct clusters showed that these components were effectively separating out the three different eye color classifications. 

![alt text](https://github.com/cse185-sp18/cse185-week5-notmaurox/blob/master/labreport/PC1andPC2Plot.png)

*Fig 1: Principal component 1 vs principal component 2 from population controlled data set*

When performing linear regression GWAS with Plink, 4 variants passed genome wide significance (p<5*10^-8 ) from the non population controlled data set and 24 variants passed from the PCA data set. Two independent signals remained after clumping and filtering out clumps that didn’t pass genome wide significance test from the PCA data set. The first clump contained two SNPs and the second contained 12. In determining which data set to use for clumping, population controlled (PCA) vs non-population controlled were compared using the QQ and Manhattan plots reported below. 

![alt text]( https://github.com/cse185-sp18/cse185-week5-notmaurox/blob/master/labreport/withoutCovariantControlled_manhattan.png)
*Fig 2: Manhattan plot for Plink linear regression on data without controlling for population structure*

![alt text]( https://github.com/cse185-sp18/cse185-week5-notmaurox/blob/master/labreport/withoutCovariantControlled_qq.png)
*Fig 3: QQ plot for Plink linear regression on data without controlling for population structure*

![alt text]( https://github.com/cse185-sp18/cse185-week5-notmaurox/blob/master/labreport/covariantControlled_manhattan.png)
*Fig 4: Manhattan plot for Plink linear regression on PCA data with controlling for population structure*

![alt text]( https://github.com/cse185-sp18/cse185-week5-notmaurox/blob/master/labreport/covariantControlled_qq.png)
*Fig 5: QQ plot for Plink linear regression on PCA data with controlling for population structure*

The two clumps identified by plink are produced in Table 1. Of the six SNPs previously correlated with eye color, only rs12913832 ended up in one of the provided clumps. Using UCSC genome browser on hg38 assembly, rs12913832 was identified as falling in the OCA2 gene at chr15:27754875-28099358. Searching for other SNPs from the two clumps did not reveal any that fell within genes. 

| Index Variant | Clump P-value | Other SNPs in clump |
| ------------- | ------------- | ------------------- |
| rs1129038  | 2.75e-13 | rs12913832(1) |
| rs1470608 | 3.56e-08 | rs749846(1),rs3794604(1),rs4778232(1),rs1448485(1),rs16950821(1),rs8024968(1),rs7177686(1),rs6497253(1),rs7170869(1),rs1375164(1),rs6497254(1) |

*Table 1: Clumps identified by Plink that pass the global significance test*

### Eye color prediction

In supplementary data SupplementaryDataEyeColorModelCalculations.xlsx, [also available here,](https://docs.google.com/spreadsheets/d/1PQV8Cucgs4Oxt0j9pONhMgkecK3zBQnHUkPHsRb7DxI/edit?usp=sharing) the results of eye color prediction modeling on data set 2 is presented as described in the methods in `EyeColorCalculation` sheet. General statistics about each population are reported in the `PredictionByPop` sheet. Amongst the northern Europe group (CEU), the probability of having blue eyes was 0.63, brown eyes 0.24, and other 0.13. Amongst Southern Europeans (TSI group), the probability of having blue eyes was 0.25, brown 0.16, and other 0.59. **Extra Credit:** In the case of predicting the eye color of a child given the mother and fathers genotype for the 6 SNPs of interest, this data can be found on `Sheet5`. The mother and fathers genotypes only differ at the first, third, and fifth SNPs in terms of both being not heterozygous for the same allel. This meant that the child was guaranteed to have CC/GG/CC alleles for the second, fourth, and sixth SNPs. These alleles stayed constant and all variations of his genomes based on changing allele combinations he could have from his parents were written down in rows 25-36. For these varying alleles in his genotype, the frequency of each one was calculated and multiplied by the probability they resulted in blue eyes. The sum of these probabilities represented the total probability the child inherits blue eyes, 0.54.  

UCSC genome browser was used to view the 6 target SNPs from data set 2 in human reference hg19. The data provided for SNP rs12913832 is presented in figure 6. This SNP falls in an intron of the HERC2 gene but is also upstream of the OCA2 gene. Based off this information, it seems like this is a common location for histone modification H3K36m3. This histone modification has been classified as gene regulator being connected to decreasing transcription in context of RNAPII[7]. In this case, it may be decreasing the activity of OCA2, a gene that is associated in the creation of Melanin, a substance that determines skin, hair, and eye color [6]. It makes sense that a SNP falling in a potential regulatory region of this gene is being associated with eye color. More notes on SNP data from UCSC genome browser can be seen in lab notebook. 

![alt text](https://github.com/cse185-sp18/cse185-week5-notmaurox/blob/master/labreport/rs12913832UCSCGenomeBrowserScreenCap.png)
*Fig 6: Screen capture of UCSC genome browser viewing SNP rs12913832 in human reference genome hg19*

## Discussion
Controlling for population structure resulted in tighter, more precise data. Fig 2 and 3 show the Manhattan and QQ plots for data without controlling for population structure while 4 and 5 show these plots for data resulting from PCA. Fig. 2 has much more noise in p-values of SNPs identified; the spread of SNPs shows that in this data set, it was hard for Plink to separate out the significant from no significant variants. In the case of the PCA data in figure 4, the Manhattan plot becomes more even where its clear which SNPs had higher significance from the rest due to the loss of noise. In the case of the QQ plots, a good signal is one where most points fall on the line and a few significant variants jump above it. In figure 3, too many too many SNPs are inflated above the line meaning that too many points were labeled as having significant distributions. Figure 5 provides a much better picture. However, the distribution drops below the line. This might be because our data sets were not complete enough to cover all SNPs accounting for eye color. That being said, it identified fewer SNPs as significant so the results were understood as more precise and accurate. While after controlling for population structure, the amount of significant variants increased, (4 to 24) these variants were more clearly identifiable from others. 

Results from GWAS were only able to identify 1 SNP known for contributing to eye color, rs12913832. This might be a result of the data not being complete enough. As described above, the QQ plot for population structure corrected data dipped below the trend line. This is an indication that there might not be enough data for analysis. The other 5 SNPs might not have been present on the SNP genotype array used, could have suffered from poor quality scores, or may have blended into the noise that was cut out during PCA. These potential shortcomings could be addressed by using whole genome sequencing instead of SNP genotype arrays, or by using an array with different probes. The issue might also have come from computational shortcuts taken in PCA. 

There are two interesting sets of SNPs, the first being rs12913832 and rs1800407 and the second being rs12896399 and rs16891982. In the case of rs12913832 and rs1800407, rs12913832 falls upstream of OCA2 and rs1800407 falls within OCA2. Similarly, rs12896399 falls upstream of SLC24A4 and rs16891982 falls within SLC24A4. In both these cases, there is an SNP within a potential promoter region of a gene, and an SNP within the gene. This highlights OCA2 and SLC24A4 as being good candidate genes for influencing eye color. rs1393350 is on chromosome 11, in an intron of the TYR gene and rs12203592 is on chromosome 6 in an intron of IRF4. The connection these two have to eye color is less clear. They don’t agree in either being in the same gene or being in its promoter region. However, based on the analysis done, OCA2 and SLC24A4 should be genes of focus in future studies on eye color. It’d be interesting to see RNA-Seq data from eye tissue samples to see how the expression of these two genes alters in people with different eye colors. 

## Citations
1.	Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender D, Maller J, Sklar P, de Bakker PIW, Daly MJ & Sham PC (2007) PLINK: a toolset for whole-genome association and population-based linkage analysis. American Journal of Human Genetics, 81.
2.	Li H. A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data (bcftools). Bioinformatics. 2011;27(21):2987-2993. doi:10.1093/bioinformatics/btr509.
3.	UCSC Genome Browser: Kent WJ, Sugnet CW, Furey TS, Roskin KM, Pringle TH, Zahler AM, Haussler D. The human genome browser at UCSC. Genome Res. 2002 Jun;12(6):996-1006.
4.	SNPedia: a wiki supporting personal genome annotation, interpretation and analysis Michael Cariaso; Greg Lennon Nucleic Acids Research 2011; doi: 10.1093/nar/gkr798
5.	Mccarthy MI, Abecasis GR, Cardon LR, Goldstein DB, Little J, Ioannidis JPA, et al. Genome-wide association studies for complex traits: consensus, uncertainty and challenges. Nature Reviews Genetics. 2008;9(5):356–69.
6.	OCA2 gene - Genetics Home Reference [Internet]. U.S. National Library of Medicine. National Institutes of Health; [cited 2018May8]. Available from: https://ghr.nlm.nih.gov/gene/OCA2
7.	Porvair Filtration Group. Histone H3K36 [Internet]. EpiGenie | Epigenetics, Stem Cell, and Synthetic Biology News. 2014 [cited 2018May8]. Available from: http://epigenie.com/key-epigenetic-players/histone-proteins-and-modifications/histone-h3k36/
9. Walsh S, Liu F, Ballantyne KN, Oven MV, Lao O, Kayser M. IrisPlex: A sensitive DNA tool for accurate prediction of blue and brown eye colour in the absence of ancestry information. Forensic Science International: Genetics. 2011;5(3):170–80.
